src/machine/machine_nrf52xxx.go

```diff
@@ -234,23 +265,7 @@ func (spi SPI) Configure(config SPIConfig) error {
 	}
 
 	// set mode
-	switch config.Mode {
-	case 0:
-		conf &^= (nrf.SPIM_CONFIG_CPOL_ActiveHigh << nrf.SPIM_CONFIG_CPOL_Pos)
-		conf &^= (nrf.SPIM_CONFIG_CPHA_Leading << nrf.SPIM_CONFIG_CPHA_Pos)
-	case 1:
-		conf &^= (nrf.SPIM_CONFIG_CPOL_ActiveHigh << nrf.SPIM_CONFIG_CPOL_Pos)
-		conf |= (nrf.SPIM_CONFIG_CPHA_Trailing << nrf.SPIM_CONFIG_CPHA_Pos)
-	case 2:
-		conf |= (nrf.SPIM_CONFIG_CPOL_ActiveLow << nrf.SPIM_CONFIG_CPOL_Pos)
-		conf &^= (nrf.SPIM_CONFIG_CPHA_Leading << nrf.SPIM_CONFIG_CPHA_Pos)
-	case 3:
-		conf |= (nrf.SPIM_CONFIG_CPOL_ActiveLow << nrf.SPIM_CONFIG_CPOL_Pos)
-		conf |= (nrf.SPIM_CONFIG_CPHA_Trailing << nrf.SPIM_CONFIG_CPHA_Pos)
-	default: // to mode
-		conf &^= (nrf.SPIM_CONFIG_CPOL_ActiveHigh << nrf.SPIM_CONFIG_CPOL_Pos)
-		conf &^= (nrf.SPIM_CONFIG_CPHA_Leading << nrf.SPIM_CONFIG_CPHA_Pos)
-	}
+	conf = config.Mode.ApplyTo(conf)
```

Comment:
I don't feel this is necessary?
`SPIMode` type 👍 
`ApplyTo()` method 👎

Reply:
I like it because:

- It says what it does. It applies the mode to the config
- It make the switch case testable
- It hides the implementation details which are not necessary at this level of logic
- Though they are still inspect-able if one is really interested

I would like to keep it in if you have not striking counter arguments. I can also make it private if you dont like it being public.

Reply:
Test ability is no argument for you, ok. Will inline.